build-mfem/action.yml

```diff
@@ -85,9 +85,10 @@ runs:
         if [[ ${{ inputs.target }} == 'debug' ]]; then
           DEBUG="YES";
           BUILD_TYPE="Debug";
-          if [[ ${{ inputs.os }} == 'ubuntu-18.04' ]]; then
-            CPPFLAGS+=" -pedantic -Wall -Werror";
-          fi
+        fi
+        if [[ ${{ inputs.os }} != 'ubuntu-18.04' || \
```

Comment:
OK, this is what was done in travis, my bad.
However, do we really want this for the optim target too?
Also, maybe it is a good idea to keep this on the linux target, because macos resource is limited...

Reply:
> do we really want this for the optim target too?

I don't think it hurts to add these flags for the opt build -- at least with clang (which is the compiler on mac). With gcc (on linux) adding these flags to the `-O3` flag results in more warnings (compared to no optimization flags) that can be quite different from one version of gcc to another -- that's why the warning flags are disabled on linux with opt build.

> Also, maybe it is a good idea to keep this on the linux target, because macos resource is limited...

Do you think adding these flags makes the compilation slower? I'd be surprised if there is any noticeable slowdown.

Reply:
Also, I think we want this on mac because it catches issues that are not caught by gcc -- I noticed the difference with Travis because in one instance (#1492), Travis failed (due to warning failures on mac) while the github actions were clean.